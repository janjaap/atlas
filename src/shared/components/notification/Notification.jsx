import React from 'react';
import PropTypes from 'prop-types';

import './_notification.scss';

class Notification extends React.Component {
  constructor(props) {
    super(props);

    this.state = { isNotificationVisible: true };

    this.hideNotification = this.hideNotification.bind(this);
  }

  hideNotification() {
    this.setState({ isNotificationVisible: false });
  }

  render() {
    return this.state.isNotificationVisible && (
      <p className={`
        ${this.props.className}
        notification
        notification--${this.props.level}
      `}
      >
        <span className="notification__content">{this.props.children}</span>
        { this.props.level !== 'message' && this.props.canClose &&
          <button
            className="notification__button"
            onClick={this.hideNotification}
          >
            <span className="notification__button--close" />
          </button>
        }
      </p>
    );
  }
}

Notification.defaultProps = {
  canClose: true,
  className: '',
  level: 'info'
};

Notification.propTypes = {
  canClose: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
  className: PropTypes.string,
  level: PropTypes.oneOf(['alert', 'info', 'message'])
};

export default Notification;
