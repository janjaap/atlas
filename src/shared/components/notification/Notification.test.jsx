import React from 'react';
import { shallow } from 'enzyme';

import Notification from './Notification';

describe('Notification', () => {
  it('should render the provided message', () => {
    const notification = shallow(
      <Notification>
        Test message
      </Notification>
    );

    expect(notification.find('.notification__content').text()).toBe('Test message');
    expect(notification).toMatchSnapshot();
  });

  it('should add the notification level as class name', () => {
    const notification = shallow(
      <Notification level="alert">
        Test message
      </Notification>
    );

    expect(notification.find('.notification--alert').length).toBe(1);
    expect(notification).toMatchSnapshot();
  });

  it('should not show the close button for the message level', () => {
    const notification = shallow(
      <Notification level="message">
        Test message
      </Notification>
    );

    expect(notification).toMatchSnapshot();
  });

  it('should not show the close button when disabled', () => {
    const notification = shallow(
      <Notification
        level="info"
        canClose={false}
      >Test message</Notification>
    );

    expect(notification).toMatchSnapshot();
  });

  it('should be able to render `a` elements', () => {
    const notification = shallow(
      <Notification>
        <a href="http://real-address.com">External link</a>
      </Notification>
    );

    expect(notification.find('a').length).toBe(1);
    expect(notification).toMatchSnapshot();
  });

  it('should hide notification on click', () => {
    const notification = shallow(
      <Notification canClose>
        Test message
      </Notification>
    );

    notification.find('.notification__button').simulate('click');
    expect(notification.isEmptyRender()).toBe(true);
  });
});
