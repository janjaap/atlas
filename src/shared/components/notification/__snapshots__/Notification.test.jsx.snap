// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should add the notification level as class name 1`] = `
<p
  className="
        
        notification
        notification--alert
      "
>
  <span
    className="notification__content"
  >
    Test message
  </span>
  <button
    className="notification__button"
    onClick={[Function]}
  >
    <span
      className="notification__button--close"
    />
  </button>
</p>
`;

exports[`Notification should be able to render \`a\` elements 1`] = `
<p
  className="
        
        notification
        notification--info
      "
>
  <span
    className="notification__content"
  >
    <a
      href="http://real-address.com"
    >
      External link
    </a>
  </span>
  <button
    className="notification__button"
    onClick={[Function]}
  >
    <span
      className="notification__button--close"
    />
  </button>
</p>
`;

exports[`Notification should not show the close button for the message level 1`] = `
<p
  className="
        
        notification
        notification--message
      "
>
  <span
    className="notification__content"
  >
    Test message
  </span>
</p>
`;

exports[`Notification should not show the close button when disabled 1`] = `
<p
  className="
        
        notification
        notification--info
      "
>
  <span
    className="notification__content"
  >
    Test message
  </span>
</p>
`;

exports[`Notification should render the provided message 1`] = `
<p
  className="
        
        notification
        notification--info
      "
>
  <span
    className="notification__content"
  >
    Test message
  </span>
  <button
    className="notification__button"
    onClick={[Function]}
  >
    <span
      className="notification__button--close"
    />
  </button>
</p>
`;
